### PR TITLE
[ci] Make vendor/architecture definitions in benchmarks better

### PR DESCRIPTION
**Pull request by antiagainst**
This commit adds a vendor field to `_ArchitectureInfo` and
makes the readable string more intuitive.

Immediately this is to fix Mali Valhall. Mali is the overall
branding, not the microarchitecture, while Valhall is the
specific (micro)architecture.

### CODE REVIEW

File: build_tools/python/e2e_test_framework/definitions/common_definitions.py [hunk 1]
```diff
@@ -47,8 +52,8 @@ class DeviceArchitecture(_ArchitectureInfo, Enum):
   RV32_GENERIC = (ArchitectureType.CPU, "riscv_32", "generic")
 
   # Mobile GPUs
-  VALHALL_MALI = (ArchitectureType.GPU, "valhall", "mali")
-  ADRENO_GENERIC = (ArchitectureType.GPU, "adreno", "generic")
+  ADRENO_GENERIC = (ArchitectureType.GPU, "adreno", "", "qualcomm")
+  MALI_VALHALL = (ArchitectureType.GPU, "", "valhall", "arm")
```
Comment (pzread) [inline build_tools/python/e2e_test_framework/definitions/common_definitions.py]: Originally I use the format "architecture_microarchitecture" for the constant name. So if we decide to drop `mali` and use `arm` instead, should this be `VALHALL_ARM`?

Comment (pzread) [inline build_tools/python/e2e_test_framework/definitions/common_definitions.py]: If ARM is the vendor of GPU, should this be:
```suggestion
  MALI_VALHALL = (ArchitectureType.GPU, "valhall", "", "arm")
```

Comment (pzread) [reply build_tools/python/e2e_test_framework/definitions/common_definitions.py]: Ah, I see your description. In general i think we always want to have the architecture name set. If `valhall` is just a microarchitecture, is there an architecture name for it?

Note that the `architecture` here means the general ISA and `microarchitecture` is a specific hardware implementation of the ISA. For example Intel CPU has x86_64 architecture + CascadeLake microarchitecture

Comment (antiagainst) [reply build_tools/python/e2e_test_framework/definitions/common_definitions.py]: Yeah, the GPU land is messy about this actually---it's far from unified like CPU architectures where we have x86_64/ARM64 ISA to rule various impl. Instead, each GPU vendor have its own way of naming stuff. Even for the same GPU vendor, different generations of GPUs can mean vastly different ISA and chip impl, which is what microarchitecture is more concerned about. Take ARM as an example, valhall is a vastly different microarchitecture than previous generations like bifrost. The ISA corresponding to that of course changes, but GPU vendor won't tell us exactly what's the ISA and what generation it is. And there are GPU vendors like Qualcomm that basically tell us nothing about the ISA or even microarchitecture.. we just have an Adreno branding name there.. 

So I won't aim for perfection here. Just a way to make things readable.

Comment (antiagainst) [reply build_tools/python/e2e_test_framework/definitions/common_definitions.py]: Yeah, this works too, given there is a fuzzy line between the ISA and microarchitecture in various GPU vendors.. Done.

**Review by pzread — APPROVED** [review]

**Pull request merged**